test/with_dummyserver/async/test_poolmanager.py

```diff
@@ -22,3 +25,97 @@ async def test_redirect(self, http_server, backend, anyio_backend):
 
             assert r.status == 200
             assert r.data == b"Dummy server!"
+
+
+class TestFileUploads:
+    @conftest.test_all_backends
+    async def test_redirect_put_file(self, http_server, backend, anyio_backend):
+        """PUT with file object should work with a redirection response"""
+        base_url = "http://{}:{}".format(http_server.host, http_server.port)
+        retry = Retry(total=3, status_forcelist=[418])
+        # httplib reads in 8k chunks; use a larger content length
```

Comment:
Should we remove mentions of httplib?

Reply:
@pquentin Did you mean to resolve this thread?

Reply:
Not sure why my comment isn't here anymore. I think I said that it made sense to refer to httplib, but I'm not sure anymore. 🤷 But year we can keep this resolved.